2023 카카오 블라인드 채용/이모티콘 할인행사/PGS_이모티콘할인행사_김주한.cpp

```diff
@@ -0,0 +1,76 @@
+#include <string>
+#include <vector>
+#include <algorithm>
+using namespace std;
+
+int pct[4] = { 40, 30, 20, 10 };
+vector<int> path; // 이모티콘 수만큼 pct 조합
+struct NODE {
+    int isJoinNum;
+    int price;
+};
+vector<vector<int>> userList;
+vector<int> emoticonList;
+vector<NODE> v;
+
+bool cmp(NODE left, NODE right) {
+    if (left.isJoinNum > right.isJoinNum) return true;
+    if (left.isJoinNum < right.isJoinNum) return false;
+
+    if (left.price > right.price) return true;
+    if (left.price < right.price) return false;
+
+    return false;
+}
```

Comment:
```suggestion
bool cmp(NODE left, NODE right) {
    if (left.isJoinNum > right.isJoinNum) return true;
    if (left.price > right.price) return true;
    
    return false;
}
```

`left.isJoinNum <= right.isJoinNum`이거나 `left.price <= right.price`일 때는 false를 반환하니까 이렇게 간략화할 수 있을것같아요! :)